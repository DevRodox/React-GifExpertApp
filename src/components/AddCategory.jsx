import { useState } from "react";


export const AddCategory = ({ onNewCategory }) => {
    
    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();
        if( inputValue.trim().length <= 1 ) return;
        
        // setCategories( categorie => [inputValue, ...categorie]);
        onNewCategory( inputValue );
        setInputValue('');
    };

    return(
        <form onSubmit={ onSubmit }>
            <input 
                placeholder="Buscar Gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    );
};