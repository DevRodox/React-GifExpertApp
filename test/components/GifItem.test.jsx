import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {

    const title = 'One Punch' 
    const url   = 'https://media4.giphy.com/media/3ohzdOXZARMFbaqch2/giphy.gif?cid=9205ea0djz2iga5eje4jiq2xc6i9mrv4tqvd506j952xfrb3&ep=v1_gifs_search&rid=giphy.gif&ct=g'


    test('GifItem debe coincidir con el snapshot', () => {
        const { container } = render(
            <GifItem 
                title= { title }
                url= { url }
             />
        );

        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el url y el alt indicado', () => {
         
        render( <GifItem title= { title } url= { url } /> );

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect ( alt ).toBe( title );
    });

    test('Debe de mostar el titulo en el componente', () => {

        render( <GifItem title= { title } url= { url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();
    });
});