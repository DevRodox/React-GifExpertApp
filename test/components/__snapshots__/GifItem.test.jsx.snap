// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> GifItem debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://media4.giphy.com/media/3ohzdOXZARMFbaqch2/giphy.gif?cid=9205ea0djz2iga5eje4jiq2xc6i9mrv4tqvd506j952xfrb3&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
